mlflow/transformers/flavor_config.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Dict
+
+from mlflow.transformers.hub_utis import get_latest_commit_for_repo
+from mlflow.transformers.torch_utils import _extract_torch_dtype_if_set
+
+if TYPE_CHECKING:
+    import transformers
+
+
+# Flavor configuration keys
+class FlavorKey:
+    TASK = "task"
+    INSTANCE_TYPE = "instance_type"
+    TORCH_DTYPE = "torch_dtype"
+    FRAMEWORK = "framework"
+
+    MODEL = "model"
+    MODEL_TYPE = "pipeline_model_type"
+    MODEL_BINARY = "model_binary"
+    MODEL_NAME = "source_model_name"
+    MODEL_REVISION = "source_model_revision"
+
+    COMPONENTS = "components"
+    COMPONENT_NAME = "%s_name"
```

Comment:
```suggestion
    COMPONENT_NAME = "{prefix}_name"
```

can we use the `str.format` templating?

Reply:
ref: https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting

Reply:
or remove the placeholder and rename this to `XXX_SUFFIX`?

Reply:
Yeah I know the %s formatting is legacy, but key-based format was a bit verbose for short string like this `COMPONENT_NAME.format(prefix=component)` vs `COMPONENT_NAME % component`. Suffix might work better, let me give it a try.

Reply:
ah wait we can do `{}_name`